### ISSUE TITLE
pipx exits with code 1 when called from Git Bash for Windows

### ISSUE THREAD

**Issue by thecesrom** [pypa/pipx#825]
**Describe the bug**

Get exit code 1 when I run the `pipx` command on Git Bash for Windows.

```bash
$ which -a pipx
/c/Users/thecesrom/AppData/Roaming/Python/Python310/Scripts/pipx
/c/users/thecesrom/appdata/roaming/python/python310/scripts/pipx
$ pipx
$ echo $?
1
$ cat ~/.local/pipx/logs/cmd_2022-03-04_15.56.56.log
   394.8ms (setup:727): 2022-03-04 15:56:56
   394.8ms (setup:728): C:\Users\thecesrom\AppData\Roaming\Python\Python310\site-packages\pipx\__main__.py
   394.8ms (setup:729): pipx version is 1.0.0
   394.8ms (setup:730): Default python interpreter is 'C:\\Python\\Python310\\python.exe'
   400.8ms (cli:790): pipx finished.
```

The only way I can get `pipx` to run is by using `python -m pipx`.

```bash
$ python -m pipx --version
1.0.0
$ python -m pipx ensurepath
C:\Users\thecesrom\.local\bin is already in PATH.
C:\Users\thecesrom\AppData\Roaming\Python\Python310\Scripts is already in PATH.

All pipx binary directories have been added to PATH. If you are sure you want
to proceed, try again with the '--force' flag.

Otherwise pipx is ready to go!
```

Or by creating an alias (`alias pipx="python -m pipx"`).

```bash
$ alias pipx="python -m pipx"
$ pipx --version
1.0.0
```

I have tried with a clean `.bashrc` file, but it still fails.

**How to reproduce**

Pre-requisites
* Windows
* Git for Windows
* Python
* `pipx`

1. Open Git Bash
2. Run `pipx`

**Expected behavior**

`pipx` should run without errors just like it does on all other shells. `pipx` does run in PowerShell, and Command Prompt without issues.

**Comment by uranusjr** [comment]
I believe this is a problem in pip’s exe generation, see pypa/pip#10444. Sorry but this is out of pipx’s control at this time. We will try to stay on top of the situation and update accordingly.

**Comment by julie777** [comment]
I run pipx fine on windows. I only install a couple packages globally (virtualenv, virtualenvwrapper, pipx) and then use pipx to install any other tools.

How did you install pipx? 
If it was with pip v22 then the above link to pypa/pip#10444 applies and you should downgrade pip to v21
```
pip install --upgrade --force pip==21
```
and then reinstall pipx
```
pip install --upgrade --force pipx
```

The problem with pip22 causes some/many/all `<app>.exe` wrappers created during install to be defective.

**Comment by thecesrom** [comment]
I definitely installed `pipx` with `pip` 22.

As I mentioned, `pipx` runs on all shells in Windows except for Git Bash, where it only runs when `pipx` is executed as a module instead of an executable.

**Comment by julie777** [comment]
> I definitely installed `pipx` with `pip` 22.
> 
> As I mentioned, `pipx` runs on all shells in Windows except for Git Bash, where it only runs when `pipx` is executed as a module instead of an executable.

Please try downgrading pip to v21 and reinstalling pipx.

**Comment by thecesrom** [comment]
Just as an experiment I uninstalled `pipx`, and reinstalled from scratch. This meant removing all the `pipx` directory from `~/.local/pipx`, and my issue cleared on its own; now `pipx` runs as an executable on Git Bash.

## Clean install
<details>
  <summary>Click to expand</summary>

  ```bash
  $ python --version
  Python 3.10.2
  $ python -m pip freeze | xargs python -m pip uninstall -y
  ...
  $ python -m pip list
  Package    Version
  ---------- -------
  pip        22.0.4
  setuptools 60.9.3
  wheel      0.37.1
  $ python -m pip install --user pipx
  Collecting pipx
    Using cached pipx-1.0.0-py3-none-any.whl (54 kB)
  Collecting packaging>=20.0
    Using cached packaging-21.3-py3-none-any.whl (40 kB)
  Collecting userpath>=1.6.0
    Using cached userpath-1.8.0-py3-none-any.whl (9.0 kB)
  Collecting argcomplete>=1.9.4
    Using cached argcomplete-2.0.0-py2.py3-none-any.whl (37 kB)
  Collecting colorama>=0.4.4
    Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
  Collecting pyparsing!=3.0.5,>=2.0.2
    Using cached pyparsing-3.0.7-py3-none-any.whl (98 kB)
  Collecting click
    Using cached click-8.0.4-py3-none-any.whl (97 kB)
  Installing collected packages: pyparsing, colorama, argcomplete, packaging, click, userpath, pipx
  Successfully installed argcomplete-2.0.0 click-8.0.4 colorama-0.4.4 packaging-21.3 pipx-1.0.0 pyparsing-3.0.7 userpath-1.8.0
  $ python -m pip list
  Package     Version
  ----------- -------
  argcomplete 2.0.0
  click       8.0.4
  colorama    0.4.4
  packaging   21.3
  pip         22.0.4
  pipx        1.0.0
  pyparsing   3.0.7
  setuptools  60.9.3
  userpath    1.8.0
  wheel       0.37.1
  $ python -m pipx ensurepath
  C:\Users\thecesrom\.local\bin is already in PATH.
  C:\Users\thecesrom\AppData\Roaming\Python\Python310\Scripts is already in PATH.
  
  ⚠️  All pipx binary directories have been added to PATH. If you are sure you want to proceed, try again with the '--force' flag.  
  
  Otherwise pipx is ready to go! ✨ 🌟 ✨
  $ cd ~/.local/pipx/
  $ ls -al
  total 4
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 ./
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 ../
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 .cache/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 logs/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 venvs/
  $ ls -al venvs
  total 0
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 ./
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:45 ../
  ```
</details>

## `pipx`-installed packages
<details>
  <summary>Click to expand</summary>

  So only `pipx` is installed globally, and these are the packages I install using `pipx`:
  
  Packages:
  - black
  - flake8
  - isort
  - pre-commit
  - pydocstyle[toml]
  - pylint
  - pyproject-fmt
  - setup-cfg-fmt
  - sort-all
  
  ```bash
  $ ls -al ~/.local/pipx/venvs
  total 20
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:58 ./
  drwxr-xr-x 1 thecesrom  1049089 0 Mar  7 10:57 ../
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:57 black/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:57 flake8/
  drwxr-xr-x 1 thecesrom  1049089 0 Mar  7 10:57 isort/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:57 pre-commit/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:57 pydocstyle/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:58 pylint/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:58 pyproject-fmt/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:58 setup-cfg-fmt/
  drwxr-xr-x 1 thecesrom 1049089 0 Mar  7 10:58 sort-all/
  ```
  
  And all packages seem to be working fine. So a clean install fixed it for me using `pip 22.0.4`,
</details>

## Conclusion

So the first time I installed `pipx` was when Python 3.10.2 was released (2022-01-14) which came with `pip  21.2.4` according to the [Release Notes](https://docs.python.org/release/3.10.2/whatsnew/changelog.html#python-3-10-2-final), and [bpo-45343](https://bugs.python.org/issue45343).

And then I reinstalled `pipx` without cleaning the `~/.local/pipx` directory, if it even matters.

So I'm guessing that it is not entirely a `pip`-related issue, since I am using the latest version (`22.0.4`), and `pipx` and all `pipx`-installed packages seem to be working fine after a clean installation.

Maybe this could shed some light.

Thanks!

**Comment by julie777** [comment]
I was having all the problems with pip 22.0.3.
I just checked and pip22.0.4 downgraded distlib to  0.3.3 which undos the distlib change that broke the wrappers generated on Windows.
